Source/Utils/WarpXMovingWindow.cpp

```diff
@@ -394,6 +402,33 @@ WarpX::shiftMF (MultiFab& mf, const Geometry& geom, int num_shift, int dir,
             dstfab(i,j,k,n) = srcfab(i+shift.x,j+shift.y,k+shift.z,n);
         })
     }
+
+#ifdef WARPX_DIM_RZ
+    if (WarpX::GetInstance().getPMLRZ()) {
+        // This does the exchange of data in the corner guard cells, the cells that are in the
+        // guard region both radially and longitudinally. These are the PML cells in the overlapping
+        // longitudinal region. FillBoundary normally does not update these cells.
+        // This update is needed so that the cells at the end of the FABs are updated appropriately
+        // with the data shifted from the nieghboring FAB. Without this update, the RZ PML becomes
+        // unstable in with the moving grid.
+        // This code creates a temporary MultiFab using a BoxList the has all of its boxes grown
+        // to include the guard cells, so that the guard cells become effictively valid cells.
+        // The temporary MultiFab is setup to refer to the data of the original Multifab (this can
+        // be done since the shape of the data is all the same, just the indexing is different).
+        BoxList bl;
+        for (int i = 0, N=ba.size(); i < N; ++i) {
+            bl.push_back(amrex::grow(ba[i], 0, mf.nGrowVect()[0]));
```

Comment:
Just so I understand correctly: all the boxes here are grown radially using the radial number of guard cells, and then the MultiFab `rmf` created below is also grown longitudinally using a number of guard cells that is setup as `(0,longitudinal number of guard cells)`. Am I understanding this correctly?

Reply:
The size of the boxes is increased radially so that the radial guard cells are included in the valid domain. When the `rmf` is created, that `IntVect` is setting the number of guard cells (and doesn't affect what's in the valid domain). I've update the comment to make this more clear.